lib/ansible/module_utils/network/iosxr/config/interfaces/interfaces.py

```diff
@@ -0,0 +1,277 @@
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+
+        for each in have:
+            for interface in want:
+                if each['name'] == interface['name']:
+                    break
+                elif interface['name'] in each['name']:
+                    break
+            else:
+                # We didn't find a matching desired state, which means we can
+                # pretend we recieved an empty desired state.
+                interface = dict(name=each['name'])
+                kwargs = {'want': interface, 'have': each}
+                commands.extend(Interfaces._clear_config(**kwargs))
+                continue
+            have_dict = filter_dict_having_none_value(interface, each)
+            kwargs = {'want': {}, 'have': have_dict}
+            commands.extend(Interfaces._clear_config(**kwargs))
+            kwargs = {'want': interface, 'have': each, 'commands': commands}
+            commands.extend(Interfaces._set_config(**kwargs))
+        # Remove the duplicate interface call
+        commands = remove_duplicate_interface(commands)
+
+        return commands
+
+    @staticmethod
+    def _state_merged(want, have):
+        """ The command generator when state is merged
+        :rtype: A list
+        :returns: the commands necessary to merge the provided into
+                  the current configuration
+        """
+        commands = []
+
+        for interface in want:
+            for each in have:
+                if each['name'] == interface['name']:
+                    break
+                elif interface['name'] in each['name']:
+                    break
+            else:
+                continue
+            kwargs = {'want': interface, 'have': each}
+            commands.extend(Interfaces._set_config(**kwargs))
+
+        return commands
+
+    @staticmethod
+    def _state_deleted(want, have):
+        """ The command generator when state is deleted
+        :rtype: A list
+        :returns: the commands necessary to remove the current configuration
+                  of the provided objects
+        """
+        commands = []
+
+        if want:
+            for interface in want:
+                for each in have:
+                    if each['name'] == interface['name']:
+                        break
+                    elif interface['name'] in each['name']:
+                        break
+                else:
+                    continue
+                interface = dict(name=interface['name'])
+                kwargs = {'want': interface, 'have': each}
+                commands.extend(Interfaces._clear_config(**kwargs))
+        else:
+            for each in have:
+                kwargs = {'want': {}, 'have': each}
+                commands.extend(Interfaces._clear_config(**kwargs))
+
+        return commands
+
+    @staticmethod
+    def _set_config(**kwargs):
```

Comment:
Better to define an API instead of  `*kwargs` since the method is called from within the class